redis/commands/json/commands.py

```diff
@@ -213,6 +216,47 @@ def set(self, name, path, obj, nx=False, xx=False, decode_keys=False):
             pieces.append("XX")
         return self.execute_command("JSON.SET", *pieces)
 
+    def set_file(self, name, path, file_name, nx=False, xx=False, decode_keys=False):
+        """
+        Set the JSON value at key ``name`` under the ``path`` to the content
+        of the json file ``file_name``.
+
+        ``nx`` if set to True, set ``value`` only if it does not exist.
+        ``xx`` if set to True, set ``value`` only if it exists.
+        ``decode_keys`` If set to True, the keys of ``obj`` will be decoded
+        with utf-8.
+
+        """
+
+        with open(file_name, "r") as fp:
+            file_content = loads(fp.read())
+
+        return self.set(name, path, file_content, nx, xx, decode_keys)
```

Comment:
Since there are named arguments in json.set, we should pass arguments for this function upstream, accordingly.

```sug
        return self.set(name, path, file_content, nx=nx, xx=xx, decode_keys=decode_keys)
```